### PR TITLE
Region test

### PR DESCRIPTION
**Pull request by dangodev**
## Reason for change

We had a `regions` prop in `<manifold-plan-selector>` that was in our docs, but there was no test for it. This adds a test.

## Testing

Test should pass.

## Checklist

- [x] **CHANGELOG**: The **Unreleased** section of CHANGELOG was updated
- [x] **Prop changes**: [docs][docs] were updated
- [x] **Prop changes**: E2E tests were updated, testing from the highest level possible
- [x] **Platform testing**: If this change should be [tested against a platform][platform-testing], has it been?

[docs]: https://ui.sandbox.manifold.co
[platform-testing]: https://app.gitbook.com/@manifold/s/engineering/playbooks/testing-your-code-against-a-platform

### CODE REVIEW

File: stories/catalog.stories.js [hunk 1]
```diff
@@ -39,10 +39,13 @@ storiesOf('Catalog', module)
...
+    const regions = text('regions', '');
     return `
-    <manifold-plan-selector product-label="${productLabel}" ${
-      freePlans ? 'free-plans' : ''
-    }></manifold-plan-selector>`;
+    <manifold-plan-selector
+      ${freePlans ? 'free-plans' : ''}
+      product-label="${productLabel}"
+      regions="${regions}"
+    ></manifold-plan-selector>`;
```
Comment (dangodev) [inline stories/catalog.stories.js]: Adds Storybook test

File: src/data/plan-details-query.ts [hunk 2]
```diff
@@ -52,7 +52,7 @@ export default gql`
       }
     }
   }
-  regions(first: 25) {
+  regions(first: 25, orderBy: { field: DISPLAY_NAME, direction: ASC }) {
```
Comment (dangodev) [inline src/data/plan-details-query.ts]: We weren’t sorting regions before. Now we are!

File: src/components/manifold-select/manifold-select.tsx [hunk 3]
```diff
@@ -5,8 +5,8 @@ import loadMark from '../../utils/loadMark';
 
 @Component({
   tag: 'manifold-select',
-  styleUrl: 'mf-select.css',
-  scoped: true,
+  styleUrl: 'manifold-select.css',
+  shadow: true,
```
Comment (dangodev) [inline src/components/manifold-select/manifold-select.tsx]: I think I remember this—we were testing `scoped` wayy back. `shadow` is more standardized, and it makes testing easier. There’s no downside to this.

File: src/components/manifold-plan-selector/manifold-plan-selector.tsx [hunk 4]
```diff
@@ -162,17 +156,16 @@ export class ManifoldPlanSelector {
...
 
-  parseRegions(regions: string) {
-    return regions.split(',').map(region => region.trim().toLowerCase());
-  }
-
   @logger()
   render() {
+    const regions =
+      (this.regions && this.regions.split(',').map(region => region.trim())) || undefined;
+
```
Comment (dangodev) [inline src/components/manifold-plan-selector/manifold-plan-selector.tsx]: Getting rid of `Watch` simplifies this component and reduces the possibility for error. The test will catch if this logic here isn’t working.

File: src/components/manifold-plan-selector/manifold-plan-selector.tsx [hunk 5]
```diff
@@ -59,7 +59,6 @@ const resourceQuery = gql`
 
 @Component({ tag: 'manifold-plan-selector' })
 export class ManifoldPlanSelector {
-  @Element() el: HTMLElement;
```
Comment (dangodev) [inline src/components/manifold-plan-selector/manifold-plan-selector.tsx]: These were added way back in the state tunnel days. If it’s not apparent you need `@Element`, you don’t.

File: src/components/manifold-plan-details/manifold-plan-details.spec.ts [hunk 6]
```diff
@@ -50,7 +55,7 @@ describe('<manifold-plan-details>', () => {
     });
 
     it('change', async () => {
-      const { page, component } = await setup();
+      const { page, component } = await setup({});
```
Comment (dangodev) [inline src/components/manifold-plan-details/manifold-plan-details.spec.ts]: I initially added the regions test here, but then moved it higher up to test from the top level. These changes are just noise now, but they do make it easier to add more tests to this file, so might just keep it.

File: codecov.yml [hunk 7]
```diff
@@ -0,0 +1,8 @@
+ignore:
+  - docs/
+  - scripts/
+  - src/spec
+  - stories/
+  - *-happo.ts
+  - *.spec.ts
+  - *.e2e.ts
```
Comment (dangodev) [inline codecov.yml]: So I just noticed that coverage went _down_ after adding a mock and I have a sneaking suspicion we’re not ignoring enough in codecov

File: src/data/plan-details-query.ts (hunk 2 above)
Comment (domenicrosati) [reply src/data/plan-details-query.ts]: hooray

File: src/components/manifold-select/manifold-select.tsx (hunk 3 above)
Comment (domenicrosati) [reply src/components/manifold-select/manifold-select.tsx]: :shadow:

File: src/components/manifold-plan-selector/manifold-plan-selector.tsx (hunk 4 above)
Comment (domenicrosati) [reply src/components/manifold-plan-selector/manifold-plan-selector.tsx]: lgtm

**Pull request merged**